addons/html_editor/static/src/core/split_plugin.js

```diff
@@ -323,6 +347,98 @@ export class SplitPlugin extends Plugin {
+            if (!block?.isContentEditable) {
+                continue;
+            }
+
+            // Check if we can split at this line break.
+            const unsplittable = ancestors(br, block).find(this.isUnsplittable.bind(this));
+            const canWrapInUnsplittable = allowsParagraphRelatedElements(unsplittable);
+            if (unsplittable) {
+                if (canWrapInUnsplittable) {
+                    // If splitting here would split an unsplittable element,
+                    // remove the line break and wrap the content around it in
+                    // new base containers.
+                    const cursors = this.dependencies.selection.preserveSelection();
+                    const children = [...unsplittable.childNodes];
+                    const brIndex = childNodeIndex(br);
+                    // Wrap only until the first node that can't be wrapped, in
+                    // both directions.
+                    const isInvalid = (node) => !isPhrasingContent(node);
+                    const startInvalid = children.slice(0, brIndex).findLast(isInvalid);
+                    const endInvalid = children.slice(brIndex + 1).find(isInvalid);
+                    const childrenToInsert = children.slice(
+                        startInvalid ? childNodeIndex(startInvalid) + 1 : 0,
+                        endInvalid ? childNodeIndex(endInvalid) : children.length
+                    );
+                    // The new base container will become the new block parent
+                    // of the BR.
+                    block = this.dependencies.baseContainer.createBaseContainer();
+                    childrenToInsert[0]?.before(block);
+                    block.append(...childrenToInsert);
+                    cursors.restore();
+                } else {
+                    // If we can't insert a base container here, there's nothing
+                    // we can do.
+                    continue;
+                }
+            }
+
+            // Now let's split at the line break.
+            const cursors = this.dependencies.selection.preserveSelection();
+            let { anchorNode, anchorOffset, focusNode, focusOffset } = getEditableSelection();
+            const brIndex = childNodeIndex(br);
+            const oldParent = br.parentElement;
+            const [before, after] = this.splitElementUntil(oldParent, brIndex, block.parentElement);
+            br.remove();
+            [before, after].forEach(fillEmpty);
+
+            // Restore the selection.
+            if ([anchorNode, focusNode].some((node) => node === oldParent)) {
+                // We can't use `cursors.remapNode` here because the old parent
+                // was split into two new nodes.
+                // eg, `<p>[a<br>b]</p>` with selection (p, 0, p, 3) ->
+                // `<p>[a</p><p>b]</p>` -> anchor p !== focus p.
+                const remapPos = (i) => (i <= brIndex ? [before, 0] : [after, i - brIndex - 1]);
```

Comment:
What is `remapPos` mean to stand for ? Remap piece of shit ? 👀 

In any case it's not clear to me. Probably need a better name.